tests/integration_python/test_run_cli.py

```diff
@@ -1356,3 +1357,131 @@ def test_task_caching_with_multiple_inputs_args(pixi: Path, tmp_pixi_workspace:
+    windows_toml = f"""
+    [workspace]
+    name = "test"
+    channels = ["{dummy_channel_1}"]
+    platforms = ["linux-64", "osx-64", "osx-arm64", "win-64"]
+    [activation.env]
+    MY_ENV = "test123"
+    [target.unix.activation]
+    scripts = ["env_setup.sh"]
+    [target.win-64.activation]
+    scripts = ["env_setup.bat"]
+    [tasks.task]
+    cmd = "echo $env:MY_ENV"
+    [tasks.foo]
+    cmd = "echo $env:MY_ENV"
+    [tasks.foobar]
+    cmd = "echo $env:FOO_PATH"
+    [tasks.task.env]
+    MY_ENV = "test456"
+    [dependencies]
+    pixi-foobar = "*"
+    """
+    if platform.system() == "Windows":
+        manifest.write_text(windows_toml)
+        script_manifest.write_text("""
+        $env:MY_ENV = "activation_script"
+        $env:FOO_PATH = "activation_script"
+        """)
+    else:
+        manifest.write_text(unix_toml)
+        script_manifest.write_text("""
+        #!/bin/bash
+        export MY_ENV="activation_script"
+        export FOO_PATH="activation_script"
+        """)
+
+    # Run the default task
+    verify_cli_command(
+        [pixi, "run", "--manifest-path", manifest, "task"],
+        stdout_contains="test456",
+    )
+
+    # Validate that without experimental it does not use the cache
+    assert not tmp_pixi_workspace.joinpath(".pixi/activation-env-v0").exists()
+
+    # Enable the experimental cache config
+    verify_cli_command(
+        [
+            pixi,
+            "config",
+            "set",
+            "--manifest-path",
+            manifest,
+            "--local",
+            "experimental.use-environment-activation-cache",
+            "true",
+        ],
+    )
+
+    # Test 1: task.env > outside environment variable - should use environment variable defined in specific tasks
+    verify_cli_command(
+        [pixi, "run", "--manifest-path", manifest, "task"],
+        stdout_contains="test456",
+        env={"MY_ENV": "outside_ev"},
+    )
+
+    # Test 2: task.env > activation.env - should use environment variable defined in specific tasks
+    verify_cli_command(
+        [pixi, "run", "--manifest-path", manifest, "task"],
+        stdout_contains="test456",
+    )
+
+    # Test 3: activation.env > outside environment variable - should use activation.env
+    verify_cli_command(
+        [pixi, "run", "--manifest-path", manifest, "foo"],
+        stdout_contains="test123",
+        env={"MY_ENV": "outside_ev"},
+    )
+
+    # Test 4: activation.env > activation.script - should use activation.env
+    verify_cli_command(
+        [pixi, "run", "--manifest-path", manifest, "foo"],
+        stdout_contains="test123",
+    )
+
+    # Test 5: activation.script > activation scripts from dependencies - should use activation.script
+    # Run an actual install
+    verify_cli_command(
+        [pixi, "install", "--manifest-path", manifest],
+    )
+    verify_cli_command(
+        [pixi, "run", "--manifest-path", manifest, "foobar"],
+        stdout_contains="activation_script",
+    )
```

Comment:
I am missing tests here, that environment variables that are set from the outside and are NOT overridden by anything inside Pixi still work as expected

Reply:
I would also choose the tests a bit different.

So you have this goal:

1) outside environment variables
1)   activation scripts of dependencies
1)   activation.scripts
1)   activation.env
1)   task specific envs like tasks.start = {cmd = "…", env={ENV_VAR="SOME"}}

So add tests that 5>4, 4>3, 3>2, 2>1 and finally that 1 on its own works
Then you have everything covered

Reply:
I want you to add asserts that "activation scripts of dependencies" wins over "outside environment variables". 
"activation.scripts" should win over "activation scripts of dependencies", and so on. If nothing else is set "outside environment variables" should be considered.